ultralytics/solutions/analytics.py

```diff
@@ -35,20 +35,20 @@ def __init__(self, **kwargs):
         self.clswise_count = {}  # dictionary for classwise counts
 
         # Ensure line and area chart
-        if self.type in {"line", "area"}:
+        if self.analytics_type in {"line", "area"}:
             self.lines = {}
             self.fig = Figure(facecolor=self.bg_color, figsize=figsize)
             self.canvas = FigureCanvas(self.fig)  # Set common axis properties
             self.ax = self.fig.add_subplot(111, facecolor=self.bg_color)
-            if self.type == "line":
+            if self.analytics_type == "line":
                 (self.line,) = self.ax.plot([], [], color="cyan", linewidth=self.line_width)
-        elif self.type in {"bar", "pie"}:
+        elif self.analytics_type in {"bar", "pie"}:
             # Initialize bar or pie plot
             self.fig, self.ax = plt.subplots(figsize=figsize, facecolor=self.bg_color)
             self.canvas = FigureCanvas(self.fig)  # Set common axis properties
             self.ax.set_facecolor(self.bg_color)
             self.color_mapping = {}
-            self.ax.axis("equal") if type == "pie" else None  # Ensure pie chart is circular
```

Comment:
@RizwanMunawar it's fine to use self.type, this is not protected, it looks like you just made a mistake here and used the builtin `type` instead?